src/intervals/arithmetic/power.jl

```diff
@@ -0,0 +1,286 @@
+    a = Interval{Float64}(a.lo.num/a.lo.den, a.hi.num/a.hi.den)
+    return F(a^x)
+end
+
+# Rational power
+function ^(a::F, x::Rational) where {F<:Interval}
+    domain = F(0, Inf)
+
+    p = x.num
+    q = x.den
+
+    isempty(a) && return emptyinterval(a)
+    iszero(x) && return one(a)
+
+    if isthinzero(a)
+        x > zero(x) && return zero(a)
+        return emptyinterval(a)
+    end
+
+    x == (1//2) && return sqrt(a)
+
+    if x >= 0
+        if a.lo ≥ 0
+            isinteger(x) && return a ^ Int64(x)
+            a = @biginterval(a)
+            ui = convert(Culong, q)
+            low = BigFloat()
+            high = BigFloat()
+            ccall((:mpfr_rootn_ui, :libmpfr) , Int32 , (Ref{BigFloat}, Ref{BigFloat}, Culong, MPFRRoundingMode) , low , a.lo , ui, MPFRRoundDown)
+            ccall((:mpfr_rootn_ui, :libmpfr) , Int32 , (Ref{BigFloat}, Ref{BigFloat}, Culong, MPFRRoundingMode) , high , a.hi , ui, MPFRRoundUp)
+            b = interval(low, high)
+            b = convert(F, b)
+            return b^p
+        end
+
+        if a.lo < 0 && a.hi ≥ 0
+            isinteger(x) && return a ^ Int64(x)
+            a = a ∩ F(0, Inf)
+            a = @biginterval(a)
+            ui = convert(Culong, q)
+            low = BigFloat()
+            high = BigFloat()
+            ccall((:mpfr_rootn_ui, :libmpfr) , Int32 , (Ref{BigFloat}, Ref{BigFloat}, Culong, MPFRRoundingMode) , low , a.lo , ui, MPFRRoundDown)
+            ccall((:mpfr_rootn_ui, :libmpfr) , Int32 , (Ref{BigFloat}, Ref{BigFloat}, Culong, MPFRRoundingMode) , high , a.hi , ui, MPFRRoundUp)
+            b = interval(low, high)
+            b = convert(Interval{T}, b)
+            return b^p
+        end
+
+        if a.hi < 0
+            isinteger(x) && return a ^ Int64(x)
+            return emptyinterval(a)
+        end
+
+    end
+
+    if x < 0
+        return inv(a^(-x))
+    end
+end
+
+# Interval power of an interval:
+function ^(a::F, x::Interval) where {F<:Interval{BigFloat}}
+    T = BigFloat
+    domain = F(0, Inf)
+
+    a = a ∩ domain
+
+    (isempty(x) || isempty(a)) && return emptyinterval(F)
+
+    lo1 = a^x.lo
+    lo2 = a^x.hi
+    lo1 = hull(lo1, lo2)
+
+    hi1 = a^x.lo
+    hi2 = a^x.hi
+    hi1 = hull(hi1, hi2)
+
+    return hull(lo1, hi1)
+end
+
+function sqr(a::Interval)
+    return a^2
+end
+
+"""
+    pow(x::Interval, n::Integer)
+
+A faster implementation of `x^n`, currently using `power_by_squaring`.
+`pow(x, n)` will usually return an interval that is slightly larger than that
+calculated by `x^n`, but is guaranteed to be a correct
+enclosure when using multiplication with correct rounding.
+"""
+function pow(x::F, n::Integer) where {F<:Interval}
+    n < 0 && return 1/pow(x, -n)
+    isempty(x) && return x
+
+    if iseven(n) && 0 ∈ x
+        return hull(zero(x),
+                    hull(Base.power_by_squaring(F(mig(x)), n),
+                        Base.power_by_squaring(F(mag(x)), n))
+            )
```

Comment:
I think `hull` works now for any number of arguments, so this can be changed (for clarity) to 
```julia
hull(zero(x), Base.power_by_squaring(F(mig(x)), n), Base.power_by_squaring(F(mag(x)), n))
```